tests/modules/suite2p_registration/test_suite2p_registration_suite2p_utils.py

```diff
@@ -0,0 +1,136 @@
+
+        # Mock function to return just the first frame as the reference. We
+        # only specifically mock the functions we absolutely need to return
+        # data or those that prevent testing.
+        def pick_initial_reference_mock(data):
+            return data[0, :, :]
+
+        # Make compute_masks return a dummy iterable to mock the code behavior.
+        def compute_masks_mock(refImg, maskSlope):
+            return [1, 2, 3, 4]
+
+        # Mock function to pretend that phase correlation ran and found
+        # offsets. Return arrays with the correct types and a range of random
+        # values.
+        def phasecorr_mock(data, cfRefImg, maxregshift, smooth_sigma_time):
+            return (self.rng.integers(-1, 1, size=self.n_frames),
+                    self.rng.integers(-1, 1, size=self.n_frames),
+                    self.rng.uniform(-1, 1, size=self.n_frames))
+
+        # Return the input frame with no shift.
+        def shift_frame(frame, dy, dx):
+            return frame
+
+        # We want this test to always run on Mocks even when suit2p is
+        # present. Hence we mock functions to return data when needed and
+        # MagicMock certain functions that are called within the code.
+        with patch("ophys_etl.modules.suite2p_registration.suite2p_utils."
+                   "pick_initial_reference",
+                   pick_initial_reference_mock), \
+             patch("ophys_etl.modules.suite2p_registration.suite2p_utils."
+                   "phasecorr",
+                   new=phasecorr_mock), \
+             patch("ophys_etl.modules.suite2p_registration.suite2p_utils."
+                   "shift_frame",
+                   new=shift_frame), \
+             patch("ophys_etl.modules.suite2p_registration.suite2p_utils."
+                   "apply_masks",
+                   new=MagicMock()), \
+             patch("ophys_etl.modules.suite2p_registration.suite2p_utils."
+                   "compute_masks",
+                   new=compute_masks_mock):
+            # Run the code with only two iterations as there is no need for
+            # more. We are just testing that the data passes through,
+            # correctly.
+            output = compute_reference(
+                frames=mock_frames,
+                niter=2,
+                maxregshift=self.ops["maxregshift"],
+                smooth_sigma=self.ops["smooth_sigma"],
+                smooth_sigma_time=self.ops["smooth_sigma_time"])
+
+        # Test that our output data is the correct shape and that it is
+        # identical to the expected output.
+        self.assertEqual(output.shape[0], self.n_frames)
+        self.assertEqual(output.shape[1], self.n_frames)
+        self.assertTrue(np.all(np.equal(output, np.load(self.mock_data_loc))))
+
+    @pytest.mark.suite2p_only
+    def test_compute_reference(self):
+        """Test that the method creates a reference image as expected using
+        suite2p."""
```

Comment:
Can you be a little more explicit about the fact that we need to make sure the output is bitwise identical to the expected output because that is the only way we will know if Suite2P has changed its implementation of `phasecorr` (or any of the other Suite2P utilities we have decided to depend upon) between versions.

I just want to make sure that whoever has to maintain this code in 2024 knows why we made this very stringent choice for testing.